packages/bigquery/system-test/bigquery.js

```diff
@@ -596,6 +596,334 @@ describe('BigQuery', function() {
         });
       });
 
+      describe('SQL parameters', function() {
+        describe('positional', function() {
+          it('should work with strings', function(done) {
+            bigquery.query({
+              query: [
+                'SELECT url',
+                'FROM `publicdata.samples.github_nested`',
+                'WHERE repository.owner = ?',
+                'LIMIT 1'
+              ].join(' '),
+              params: ['google']
+            }, function(err, rows) {
+              assert.ifError(err);
+              assert.equal(rows.length, 1);
+              done();
+            });
+          });
+
+          it('should work with ints', function(done) {
+            bigquery.query({
+              query: [
+                'SELECT url',
+                'FROM `publicdata.samples.github_nested`',
+                'WHERE repository.forks > ?',
+                'LIMIT 1'
+              ].join(' '),
+              params: [1]
+            }, function(err, rows) {
+              assert.ifError(err);
+              assert.equal(rows.length, 1);
+              done();
+            });
+          });
+
+          it('should work with floats', function(done) {
+            bigquery.query({
+              query: [
+                'SELECT snow_depth',
+                'FROM `publicdata.samples.gsod`',
+                'WHERE snow_depth >= ?',
+                'LIMIT 1'
+              ].join(' '),
+              params: [12.5]
+            }, function(err, rows) {
+              assert.ifError(err);
+              assert.equal(rows.length, 1);
+              done();
+            });
+          });
+
+          it('should work with booleans', function(done) {
+            bigquery.query({
+              query: [
+                'SELECT url',
+                'FROM `publicdata.samples.github_nested`',
+                'WHERE public = ?',
+                'LIMIT 1'
+              ].join(' '),
+              params: [true]
+            }, function(err, rows) {
+              assert.ifError(err);
+              assert.equal(rows.length, 1);
+              done();
+            });
+          });
+
+          it('should work with dates', function(done) {
+            bigquery.query({
+              query: [
+                'SELECT subject',
+                'FROM `bigquery-public-data.github_repos.commits`',
+                'WHERE author.date < ?',
+                'LIMIT 1'
+              ].join(' '),
+              params: [new Date()]
+            }, function(err, rows) {
+              assert.ifError(err);
+              assert.equal(rows.length, 1);
+              done();
+            });
+          });
+
+          it('should work with arrays', function(done) {
+            bigquery.query({
+              query: 'SELECT * FROM UNNEST (?)',
+              params: [
+                [
+                  25,
+                  26,
+                  27,
+                  28,
+                  29
+                ]
```

Comment:
Just wanting some clarification, but if an array is provided all types within said array must be the same?